### PR TITLE
Vault secret management doc additions

### PR DESCRIPTION
**Pull request by phillxnet**
Summary of proposed changes; one commit per item:

- add strip() to example vaultToken in file: avoids 'Invalid header value'.
- add secretsmount option to example vault secrets provider code.
- expand vault token docs to emphasise authorization component re policies.
- indicate buildbots 'hard wired' KV v1 and v2 vault path expectations.
- clarify connection re Vault 'apiVersion' argument and KV v1/v2 engines.
- avoid "Unknown component name: SecretInVault": via direct class name use.

Tested via suggested development venv 'make docs' and the resulting html rendered as expected in both Chrome and Firefox.

## Contributor Checklist:

* [x] I have created a file in the `master/buildbot/newsfragments` directory (and read the `README.txt` in that directory)
* [x] I have updated the appropriate documentation

### CODE REVIEW

**Comment by phillxnet** [comment]
Ready for review.
I've made the suggested spelling changes / additions suggested after my initial submission and now only see what looks like unrelated failures in the last automated tests.

Note that the original spelling issues that showed up did not surface when I ran:
```
pip install -r requirements-ci.txt
pip install -r requirements-cidocs.txt
make docs
```
locally as suggested at: https://docs.buildbot.net/latest/developer/pull-request.html#documentation

Let me know if I need to make any more changes to this pr from here on.

**Comment by phillxnet** [comment]
Linking this pr to the potentially relevant existing issue:
"secrets.SecretInVault doesn't exist" #4510
It may be that the 'proper' fix for that issue is code based, in which case I can revert the relevant change made in this pr once that code fix is in place.

File: master/docs/manual/secretsmanagement.rst [hunk 1]
```diff
@@ -119,18 +119,22 @@ SecretInVault
...
+                            secretsmount="secret",
                             apiVersion=2
     )]
 
 Vault secures, stores, and tightly controls access to secrets.
 Vault presents a unified API to access multiple backends.
-At the moment buildbot supports KV v1 and v2 backends.
-To be authenticated in Vault, Buildbot needs to send a token to the vault server.
-The token is generated when the Vault instance is initialized for the first time.
+At the moment buildbot supports KV v1 and v2 backends via the apiVersion
```
Comment (sa2ajj) [inline master/docs/manual/secretsmanagement.rst]: one-sentence-per-line

File: master/docs/manual/secretsmanagement.rst [hunk 2]
```diff
@@ -119,18 +119,22 @@ SecretInVault
...
+argument.
 
+Buildbot's Vault authentication/authorisation is via a token. The "Initial Root
```
Comment (sa2ajj) [inline master/docs/manual/secretsmanagement.rst]: ditto

File: master/docs/manual/secretsmanagement.rst [hunk 3]
```diff
@@ -230,7 +234,11 @@ Then, export the environment variable VAULT_ADDR needed to init Vault.
 Writing secrets
 ```````````````
 
-By default Vault is initialized with a mount named secret.
+By default the official docker instance of Vault is initialized with a mount path
+of 'secret', a KV v1 secret engine, and a second KV engine (v2) at 'secret/data'.
+Currently buildbot is "hard wired" to expect KV v2 engines to reside within this
+"data" sub path. Provision is made to set a top level path via the "secretsmount"
```
Comment (sa2ajj) [inline master/docs/manual/secretsmanagement.rst]: ditto

File: master/docs/spelling_wordlist.txt [hunk 4]
```diff
@@ -34,6 +35,7 @@ Atlassian
 attrs
 auth
 Auth
+authorisation
```
Comment (sa2ajj) [inline master/docs/spelling_wordlist.txt]: not a change request: british? :smile:

**Comment by phillxnet** [comment]
@sa2ajj Thanks for the review, much appreciated. I'll get to the reformatting (one sentence per line) on my next buildbot time slot. Hopefully soon.
Cheers.

**Comment by phillxnet** [comment]
Requested changes applied file wide and popped in a minor additional capitalisation issue on "Buildbot".

Initial automated test failure looks to be unrelated.

Standing by for further feedback / state change.

**Pull request merged**